File: Game Python/exo1.py
import random

player_health = 100
player_attack = 20
player_defense = 10
treasure_count = 0

enemy_health = 50
enemy_attack = 15
enemy_defense = 15

treasures = [
    "Sceptre d'Avo",
    "Potion de volonté",
    "Armure du Héros"
]

print("Bienvenue dans Fable: The Lost Chapters !")
player_name = input("Entrez votre nom de héros : ")

while player_health > 0:
    print("\n" + player_name + ", Que voulez-vous faire ?")
    print("1. Explorer les ruines")
    print("2. Combattre les bandits")
    print("3. Quitter le jeu")

    choice = int(input("Choix : "))

    if choice == 1:
        print("\nVous explorez les ruines anciennes...")

        for _ in range(3):
            treasure = random.randint(0, 1)

            if treasure:
                found_treasure = random.choice(treasures)
                print("Vous avez trouvé un trésor rare : " + found_treasure + " !")
                treasure_count += 1
            else:
                print("Vous ne trouvez rien d'intéressant...")
    elif choice == 2:
        print("\nVous affrontez un groupe de bandits...")

        while enemy_health > 0 and player_health > 0:
            enemy_health -= player_attack - enemy_defense
            player_health -= enemy_attack - player_defense

        if player_health > 0:
            print("Vous avez vaincu les bandits !")
        else:
            print("Vous avez été vaincu par les bandits...")
            break
    elif choice == 3:
        print("\nMerci d'avoir joué à Fable: The Lost Chapters ! Au revoir, " + player_name + " !")
        break

inventory = [
    "Épée de l'Élu",
    "Arc de Skorm",
    "Bouclier de l'Ombre",
    "Casque de l'Ancien",
    "Potion de résurrection"
]

print("\nVotre aventure est terminée. Vous avez trouvé", treasure_count, "trésor(s).")
print("Voici votre inventaire de héros :")

for item in inventory:
    print("-", item)

File: exo1.c
#include <stdio.h>

int main() {

    int playerHealth = 100;
    int playerAttack = 20;
    int playerDefense = 10;
    int treasureCount = 0;
    char playerName[20];
    int i;

    printf("Bienvenue dans Fable !\n");
    printf("Entrez votre nom : ");
    scanf("%s", playerName);
    

    while (playerHealth > 0) {
        printf("\n %s, Que voulez-vous faire ? \n", playerName);
        printf("\n 1. Explorer \n");
        printf("\n 2. Attaquer \n");
        printf("\n 3. Quitter \n");
        
        int choice;
        printf("Choix : ");
        scanf("%d", &choice);
        

        if (choice == 1) {
            printf("\n Vous explorez les environs... \n");
            

            for (i = 0; i < 3; i++) {
                int treasure = rand() % 2;
                
                if (treasure) {
                    printf("\n Vous avez trouvé un trésor ! \n");
                    treasureCount++;
                } else {
                    printf("\n Vous ne trouvez rien... \n");
                }
            }
        }

        else if (choice == 2) {
            printf("\n Vous attaquez un ennemi... \n");
            
            int enemyHealth = 50;
            int enemyAttack = 15;
            int enemyDefense = 15;
            
            while (enemyHealth > 0 && playerHealth > 0) {
                enemyHealth -= playerAttack - enemyDefense;
                playerHealth -= enemyAttack - playerDefense;
            }
            
            if (playerHealth > 0) {
                printf("\n Vous avez vaincu l'ennemi ! \n");
            } else {
                printf("\n Vous avez été vaincu par l'ennemi... \n");
                break;
            }
        }
        
        else if (choice == 3) {
            printf("\n Merci d'avoir joué ! Au revoir %s ! \n", playerName);
            break;
        }
    }
    
    printf("\n Votre aventure est terminée. Vous avez trouvé %d trésor. \n", treasureCount);
    
    return 0;
}
